libs/ui-components/src/components/Device/DeviceDetails/DeviceDetailsTab.tsx

```diff
@@ -68,7 +68,11 @@ const DeviceDetailsTab = ({ device, refetch, children, canEdit }: React.PropsWit
                   <StackItem className="fctl-device-details-tab__label">{t('Labels')}</StackItem>
                 </Stack>
                 <StackItem>
-                  <EditLabelsForm device={device} onDeviceUpdate={refetch} canEdit={canEdit} />
+                  {canEdit ? (
```

Comment:
Nothing would be shown if the device is non-editable and there were no labels.

Anyway, I think it's best to split the component as `EditLabelsForm` is always meant to be used for editing the labels.